pkg/cosign/fulcioverifier/ctl/verify.go

```diff
@@ -27,20 +27,43 @@ import (
 	ct "github.com/google/certificate-transparency-go"
 	ctx509 "github.com/google/certificate-transparency-go/x509"
 	"github.com/google/certificate-transparency-go/x509util"
-	"github.com/sigstore/cosign/v2/cmd/cosign/cli/fulcio/fulcioverifier/ctutil"
 	"github.com/sigstore/cosign/v2/pkg/cosign/env"
+	"github.com/sigstore/cosign/v2/pkg/cosign/fulcioverifier/ctutil"
 
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
 	"github.com/sigstore/sigstore/pkg/tuf"
 )
 
-// This is the CT log public key target name
-var ctPublicKeyStr = `ctfe.pub`
+// LogIDMetadata holds information for mapping a key ID hash (log ID) to associated data.
+type LogIDMetadata struct {
```

Comment:
could we make this the same as the rekor key map? maybe the rekor one uses ecdsa keys, but both could use a public key, or even signature.Verifier

Reply:
Yeah, thought about that :) See the 'implementation notes' in the PR description. It does make the change a bit larger (not sure what 'bit' here means yet :) ).

When you say same, I think it would be wonky to use it here and still call it TrustedRekorPubKeys for users. But, if we change the CheckOpts for RekorPubKeys to use a more generic (whatever name we should?) type that accepts crypto.PublicKey then that's a breaking change.
My preference would be to use PublicKey here instead of Verifier as a user of these methods.

Reply:
>  then that's a breaking change.

fortunately, this hasn't been in a release yet, so we're welcome to break it before 2.0

Reply:
@asraa 
https://github.com/chainguard-dev/mono/pull/7143